src/ConfigCatProvider.tsx

```diff
@@ -104,97 +107,84 @@ function serverContextNotSupported(): Error {
     throw serverContextNotSupported();
   }
-
-  getAllValuesAsync(_user?: User | undefined): Promise<SettingKeyValue<SettingValue>[]> {
+  getAllValuesAsync(_user?: User): Promise<SettingKeyValue<SettingValue>[]> {
     throw serverContextNotSupported();
   }
-
-  getAllValueDetailsAsync(_user?: User | undefined): Promise<IEvaluationDetails<SettingValue>[]> {
+  getAllValueDetailsAsync(_user?: User): Promise<IEvaluationDetails<SettingValue>[]> {
     throw serverContextNotSupported();
   }
-
   getKeyAndValueAsync(_variationId: string): Promise<SettingKeyValue<SettingValue> | null> {
     throw serverContextNotSupported();
   }
-
   forceRefreshAsync(): Promise<RefreshResult> {
     throw serverContextNotSupported();
   }
-
   waitForReady(): Promise<ClientCacheState> {
     throw serverContextNotSupported();
   }
-
   snapshot(): IConfigCatClientSnapshot {
     throw serverContextNotSupported();
   }
-
   setDefaultUser(_defaultUser: User): void {
     throw serverContextNotSupported();
   }
-
   clearDefaultUser(): void {
     throw serverContextNotSupported();
   }
-
+  isOffline: boolean;
   setOnline(): void {
     throw serverContextNotSupported();
   }
-
   setOffline(): void {
     throw serverContextNotSupported();
   }
-
-  dispose(): void { }
-
+  dispose(): void {
+    throw serverContextNotSupported();
+  }
   addListener<TEventName extends keyof HookEvents>(_eventName: TEventName, _listener: (...args: HookEvents[TEventName]) => void): this {
     throw serverContextNotSupported();
   }
-
   on<TEventName extends keyof HookEvents>(_eventName: TEventName, _listener: (...args: HookEvents[TEventName]) => void): this {
     throw serverContextNotSupported();
   }
-
   once<TEventName extends keyof HookEvents>(_eventName: TEventName, _listener: (...args: HookEvents[TEventName]) => void): this {
     throw serverContextNotSupported();
   }
-
   removeListener<TEventName extends keyof HookEvents>(_eventName: TEventName, _listener: (...args: HookEvents[TEventName]) => void): this {
     throw serverContextNotSupported();
   }
-
   off<TEventName extends keyof HookEvents>(_eventName: TEventName, _listener: (...args: HookEvents[TEventName]) => void): this {
     throw serverContextNotSupported();
   }
-
-  removeAllListeners(_eventName?: keyof HookEvents | undefined): this {
+  removeAllListeners(_eventName?: keyof HookEvents): this {
     throw serverContextNotSupported();
   }
-
   listeners(_eventName: keyof HookEvents): Function[] {
     throw serverContextNotSupported();
   }
-
   listenerCount(_eventName: keyof HookEvents): number {
     throw serverContextNotSupported();
   }
-
   eventNames(): (keyof HookEvents)[] {
     throw serverContextNotSupported();
   }
 }
 
+export function createConfigCatProviderError(methodName: string, providerId?: string): Error {
+
+  const providerIdText: string = providerId ? ` with id="${providerId}"` : "";
+
+  return Error(`${methodName} must be used in ConfigCatProvider${providerIdText}!`);
```

Comment:
I can't see too much value in it...but if you would like to highlight it what do think about this: "...ConfigCatProvider without `id` attribute" ?

Reply:
Without the addition it can be confusing as the text suggests you need to use the function in *any* ConfigCatProvider. But you can use it only in a ConfigCatProvider with no id.

So I'd like to highlight that fact. "...ConfigCatProvider without id attribute" is fine by me.